### PR TITLE
Add pools, providers and memspaces to the web docs

### PR DESCRIPTION
**Pull request by bratpiorka**
### Description

Add pools, providers, and memspaces to the web docs + small fixes.
preview: https://bratpiorka.github.io/unified-memory-framework/api.html

### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

- [x] CI workflows, not executed per PR (e.g. Nightly), execute properly 
- [ ] New tests added, especially if they will fail without my changes
- [ ] Added/extended example(s) to cover this functionality
- [x] Extended the README/documentation
- [ ] All newly added source files have a license
- [ ] All newly added source files are referenced in CMake files

### CODE REVIEW

**Review by igchor — APPROVED** [review]

**Review by kswiecicki — APPROVED** [review]

File: include/umf/providers/provider_os_memory.h
[the remark this answers is not in the document]
Comment (bratpiorka) [reply include/umf/providers/provider_os_memory.h:1]: done

**Review by kswiecicki — APPROVED** [review]

File: scripts/docs_config/api.rst [hunk 1]
```diff
@@ -3,22 +3,75 @@ API Documentation
...
+    
+Disjoint Pool
+------------------------------------------
+.. doxygenfile:: pool_disjoint.h
+    :sections: define enum typedef func var
+
+Jemalloc Pool
+------------------------------------------
+
+[jemalloc](https://github.com/jemalloc/jemalloc)-based memory pool manager.
```
Comment (lukaszstolarczuk) [inline scripts/docs_config/api.rst:27]: this link won't work in rst file

File: scripts/docs_config/api.rst [hunk 2]
```diff
@@ -3,22 +3,75 @@ API Documentation
...
 Memory Pool
-----------------------------------------------------------
+------------------------------------------
+
 .. doxygenfile:: memory_pool.h
-    :sections: enum typedef func var
+    :sections: define enum typedef func var
+    
+Disjoint Pool
+------------------------------------------
```
[the remark this answers is not in the document]
Comment (bratpiorka) [reply scripts/docs_config/api.rst:20]: Added, but note that this PR focuses on adding missing files and fixing problems, not on improving documentation.

[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply scripts/docs_config/api.rst:30]: I see, thanks anyway!

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**